runtime/meta/ffi.cpp

```diff
@@ -304,14 +304,23 @@ extern "C" {
     if (!mpz_fits_ulong_p(size)) {
       throw std::invalid_argument("Size is too large");
     }
+    if (!mpz_fits_ulong_p(align)) {
+      throw std::invalid_argument("Alignment is too large");
+    }
 
     if (allocatedKItemPtrs.find(kitem) != allocatedKItemPtrs.end()) {
       return allocatedKItemPtrs[kitem];
     }
 
     size_t s = mpz_get_ui(size);
+    size_t a = mpz_get_ui(align);
 
-    string * ret = (string *) calloc(sizeof(string *) + s, 1);
+    string * ret;
+    int result = posix_memalign((void **)&ret, a < sizeof(void *) ? sizeof(void *) : a, sizeof(string *) + s);
```

Comment:
Why not `aligned_alloc`? That's likely to also work on non-posix systems.

Reply:
Well, the llvm backend already depends on posix_memalign, but aligned_alloc requires that size is a multiple of alignment, which we don't necessarily want to assume.